packages/backend-deployer/src/cdk_deployer.ts

```diff
@@ -161,23 +173,54 @@ export class CDKDeployer implements BackendDeployer {
     // actionable errors being hidden among the stdout. Moreover execa errors are
     // useless when calling CLIs unless you made execa calling error.
     let aggregatedStderr = '';
-    const aggregatorStream = new stream.Writable();
-    aggregatorStream._write = function (chunk, encoding, done) {
+    let aggregatedStdout = '';
+    const aggregatorStderrStream = new stream.Writable();
+    const aggregatorStdoutStream = new stream.Writable();
+    aggregatorStderrStream._write = function (chunk, encoding, done) {
       aggregatedStderr += chunk;
       done();
     };
+    aggregatorStdoutStream._write = function (chunk, encoding, done) {
+      aggregatedStdout += chunk;
+      done();
+    };
     const childProcess = execa(command, cdkCommandArgs, {
       stdin: 'inherit',
-      stdout: 'inherit',
+      stdout: 'pipe',
       stderr: 'pipe',
     });
-    childProcess.stderr?.pipe(aggregatorStream);
-
+    childProcess.stderr?.pipe(aggregatorStderrStream);
+    childProcess.stdout?.pipe(aggregatorStdoutStream);
+    childProcess.stdout?.pipe(process.stdout);
     try {
       await childProcess;
+      return aggregatedStdout;
```

Comment:
I experimented something here https://github.com/aws-amplify/amplify-backend/pull/642/commits/96414ec49f8e7cfc0e51c8f91ba72d78c086b965 looks a bit clunky, wdyt?

Reply:
Why not register `stdout.on('data',...)` instead of piping?
Regardless of that there's one problem to solve, i.e. `data` or `chunk` might not be coming in a form of lines (could be partial line or many lines).

Can you try this approach to listen to stdout?
https://github.com/aws-amplify/amplify-backend/blob/79ac139975278c462be2a8196a87d47ed4e46f77/packages/integration-tests/src/process-controller/process_controller.ts#L66-L69

Reply:
Tried here https://github.com/aws-amplify/amplify-backend/pull/642/commits/ae1a81fdc2eb1f02d2c8a18d49d0ca5f23f2bd79

`on` listener does send data line by line

Reply:
> Tried here https://github.com/aws-amplify/amplify-backend/commit/ae1a81fdc2eb1f02d2c8a18d49d0ca5f23f2bd79

Looks good I think. We might want `new CDKOutputListener.listen(childProcess.stdout)`.

> on listener does send data line by line

is this documented somewhere or do we get this behavior because child process happens to emit line by line that fits pipe buffer?

Reply:
I think it's happening because the data is coming line by line, so there is a chance that a big chunk gets broken down by node. I'll look into using `readline`

Reply:
Updated here https://github.com/aws-amplify/amplify-backend/pull/642/commits/66a7ca8e24f4076499ea54465d332727c4b3e798